src/utility/components.js

```diff
@@ -493,129 +542,225 @@ export class Chant {
-      const ncSeptenary = sortedNeumeComponents.map((nc) => nc.septenary());
+
+      const lowerBoundNCSeptenary = new NeumeComponentSQ('', '', '', pitch, lowerOctaveBoundary).septenary();
+      const upperBoundNCSeptenary = new NeumeComponentSQ('', '', '', pitch, upperOctaveBoundary).septenary();
+
+      const ncSeptenary = this.neumeComponents.map((nc) => nc.septenary());
 
       const totalNotes = ncSeptenary.length;
       const totalNotesInRange = ncSeptenary.filter((septenaryValue) => septenaryValue >= lowerBoundNCSeptenary && septenaryValue <= upperBoundNCSeptenary).length;
 
       return totalNotesInRange / totalNotes;
     }
+
     let modeFromAmbitus = -1;
-    let ratingAuthentic = 0, ratingPlagal = 0;
+    let ambitusAuthenticRange, ambitusPlagalRange;
+    let ambitusAuthenticRating = 0, ambitusPlagalRating = 0;
     if (finalisPitch === 'd') {
-      ratingAuthentic = pitchRangeRate('d');
-      ratingPlagal = pitchRangeRate('a');
+      ambitusAuthenticRange = 'd';
+      ambitusPlagalRange = 'a';
     } else if (finalisPitch === 'e') {
-      ratingAuthentic = pitchRangeRate('e');
-      ratingPlagal = pitchRangeRate('b');
+      ambitusAuthenticRange = 'e';
+      ambitusPlagalRange = 'b';
     } else if (finalisPitch === 'f') {
-      ratingAuthentic = pitchRangeRate('f');
-      ratingPlagal = pitchRangeRate('c');
+      ambitusAuthenticRange = 'f';
+      ambitusPlagalRange = 'c';
     } else if (finalisPitch === 'g') {
-      ratingAuthentic = pitchRangeRate('g');
-      ratingPlagal = pitchRangeRate('d');
+      ambitusAuthenticRange = 'g';
+      ambitusPlagalRange = 'd';
     } else {
-      modeDescription += `Unable to detect the exact mode based on ambitus.\n`;
+      modeDescription += `Unable to detect the exact mode based on ambitus.`;
     }
 
-    if (ratingAuthentic > ratingPlagal) {
+    ambitusAuthenticRating = pitchRangeRate('authentic', ambitusAuthenticRange);
+    ambitusPlagalRating = pitchRangeRate('plagal', ambitusPlagalRange);
+
+    let ambitusAuthDesc = document.createElement('li');
+    ambitusAuthDesc.innerHTML = `For the authentic mode (mode ${authenticMode}): ${Number(ambitusAuthenticRating).toFixed(4) * 100}% of the notes\
+      are within the range '${ambitusAuthenticRange.toUpperCase()}-${ambitusAuthenticRange.toUpperCase()}'\
+      (${ambitusAuthenticRange.toUpperCase()}${finalisNC.getOctave()}-${ambitusAuthenticRange.toUpperCase()}${finalisNC.getOctave() + 1}).`;
+
+
+    let ambitusPlagDesc = document.createElement('li');
+    ambitusPlagDesc.innerHTML = `For the plagal mode (mode ${plagalMode}): ${Number(ambitusPlagalRating).toFixed(4) * 100}% of the notes\
+      are within the range'${ambitusPlagalRange.toUpperCase()}-${ambitusPlagalRange.toUpperCase()}'\
+      (${ambitusPlagalRange.toUpperCase()}${finalisNC.getOctave() - 1}-${ambitusPlagalRange.toUpperCase()}${finalisNC.getOctave()}).`;
```

Comment:
This needs to change based on the last commit where the lower and upper octave of the ambitus for plagal were modified.